### PR TITLE
Add Lucky Honeybadger::AuthenticHandler

### LINKED ISSUE #20: Packaged middleware for Lucky
[Lucky Framework](https://luckyframework.org/) has an out of the box authentication library called Authentic. Does it make sense to package Honeybadger-Crystal with a middleware which automatically fetches the user_id from the session?

```crystal
class Honeybadger::AuthenticMiddleware
  def initialize(@factory : Honeybadger::HttpPayload.class = Honeybadger::HttpPayload)
  end

  def call(context)
    if user_id = context.session.get(Authentic::ActionHelpers::SIGN_IN_KEY)
      Honeybadger.context(user_id: user_id)
    end

    call_next(context)
  rescue exception
    payload = @factory.new(exception, http_context.request)

    Honeybadger::Dispatch.send_async(payload)

    raise exception
  end
end
```

### PR DESCRIPTION
**Pull request by stephendolan**
I'm not sure about the structure or README changes, but thought this could be a good starting point. Equally unsure about how/if to test this programmatically, but it does work in my Lucky app 😄 .

Closes #20

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,8 +42,28 @@ end
...
+1. Add the shard to `shard.yml`
+2. Require the shard in `src/shards.cr`
+3. Add the built-in `Honeybadger::AuthenticHandler` to your middleware stack:
+
+    ```crystal
+    def middleware : Array(HTTP::Handler)
+      [
+        # ...
+        Lucky::ErrorHandler.new(action: Errors::Show),
+        Honeybadger::AuthenticMiddleware.new,
```
Comment (robacarp) [inline README.md:56]: I think this is just a late stage rename on your part, but below you have `Honeybadger::AuthenticHandler` instead of _middleware_

File: src/honeybadger/authentic_handler.cr [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#
+# It will attempt to fetch a user's ID from the current session, and include it in the error context if available.
+class Honeybadger::AuthenticHandler
+  include HTTP::Handler
+
+  def initialize(@factory : Honeybadger::HttpPayload.class = Honeybadger::HttpPayload)
+  end
+
+  def call(context : HTTP::Server::Context)
+    if (user_id = context.session.get?(Authentic::ActionHelpers::SIGN_IN_KEY))
```
Comment (robacarp) [inline src/honeybadger/authentic_handler.cr:11]: Let's add this session key to the initializer so it's more extensible.

**Review by robacarp — COMMENTED** [review]
Nice, thanks for jumping in. I left a couple nitpicky comments.

I know the HTTP::Handler packaged doesn't have any tests. I'd like to add a test around the user_id fetch portion of this at a minimum. I think you can get away with a mock which just provides context.session as a hash, and then inject whatever values you want in that hash to pass/fail the test as needed.

What do you think?

File: src/honeybadger/authentic_handler.cr (hunk 2 above)
Comment (joshuap) [reply src/honeybadger/authentic_handler.cr:11]: @stephendolan @robacarp does this still need work? lmk when you want me to merge. 👍

Comment (stephendolan) [reply src/honeybadger/authentic_handler.cr:11]: Yeah, I still need to add tests 😄 

Currently refactoring my house in my spare time doing some renovations, so it may be a minute!

**Comment by robacarp** [comment]
Closing for #23
